pkg/tsdb/grafana-testdata-datasource/testdata.go

```diff
@@ -9,9 +9,14 @@ import (
 	"github.com/grafana/grafana-plugin-sdk-go/backend/log"
 	"github.com/grafana/grafana-plugin-sdk-go/backend/resource/httpadapter"
 	"github.com/grafana/grafana-plugin-sdk-go/data"
+
+	"github.com/grafana/grafana/pkg/plugins"
 	"github.com/grafana/grafana/pkg/tsdb/grafana-testdata-datasource/sims"
 )
 
+// nolint:depguard -- this only ensures that testdata implements all client functions
+var _ plugins.Client = &Service{}
```

Comment:
we could remove this -- but kinda nice to know that we want it to fully implement client 🤷🏻 

<img width="852" alt="image" src="https://github.com/grafana/grafana/assets/705951/9649e679-5d97-4efe-9e5d-a4e609863e19">

Reply:
There's other places in the code validating this, but if you want to do it should check with SDK instead of Grafana core, i.e.

```go
var _ backend.QueryDataHandler = &Service{}
var _ backend.CallResourceHandler = &Service{}
var _ backend.CheckHealthHandler = &Service{}
```

Reply:
the issue is that plugin.Client requires all of them.  I had to add the metrics collector to use it as a client 🤷🏻